tools/downloader/topologies.py

```diff
@@ -0,0 +1,196 @@
+        '''
+        import cv2 as cv
+
+        if useIR:
+            model, config = self.getIR()
+        else:
+            model, config = self.model, self.config
+
+        if self.task_type == 'detection':
+            m = cv.dnn_DetectionModel(model, config)
+        elif self.task_type == 'classification':
+            m = cv.dnn_ClassificationModel(model, config)
+        elif self.task_type == 'semantic_segmentation':
+            m = cv.dnn_SegmentationModel(model, config)
+        else:
+            m = cv.dnn_Model(model, config)
+
+        if useIR:
+            return m
+
+        if '--mean_values' in self.mo_args:
+            mean = self.mo_args['--mean_values']
+            mean = mean[mean.find('[') + 1:mean.find(']')].split(',')
+            mean = [float(val) for val in mean]
+            m.setInputMean(mean)
+
+        if '--scale_values' in self.mo_args:
+            scale = self.mo_args['--scale_values']
+            scale = 1.0 / float(scale[scale.find('[') + 1:scale.find(']')])
+            m.setInputScale(scale)
+
+        if '--input_shape' in self.mo_args:
+            shape = self.mo_args['--input_shape']
+            shape = shape[shape.find('[') + 1:shape.find(']')].split(',')
+            shape = [int(val) for val in shape]
+            if len(shape) == 4:
+                if self.framework == 'tf':  # NHWC
+                    w, h = shape[2], shape[1]
+                else:  # NCHW
+                    w, h = shape[3], shape[2]
+            m.setInputSize(w, h)
+
+        if '--reverse_input_channels' in self.mo_args:
+            m.setInputSwapRB(True)
+
+        return m
+
+
+    def getIENetwork(self):
+        '''
+        Creates openvino.inference_engine.IENetwork instance.
+        Model Optimizer is lauched to create OpenVINO Intermediate Representation (IR).
+        '''
+        from openvino.inference_engine import IENetwork
+        xmlPath, binPath = self.getIR()
+        return IENetwork(xmlPath, binPath)
+
+
+for _topology in common.load_models(argparse.Namespace(config=None)):
+    _name = _topology.name.replace('-', '_').replace('.', '_')
+    _download_dir = _cache / _topology.subdirectory
+    _mo_args = {}
+    if _topology.mo_args:
+        for _arg in _topology.mo_args:
+            _tokens = _arg.split('=')
+            _mo_args[_tokens[0]] = _tokens[1] if len(_tokens) == 2 else None
+
+    # Resolve weights and text files names.
+    if _topology.framework == 'dldt':
+        assert(len(_topology.files) >= 2), 'Expected to have at least two files in IR format'
+        _paths = []
+        # DLDT topologies usually come with multiple precision configuration files.
+        for i in range(2):
+            # Get only basename and add precision marker
+            _path = str(_topology.files[i].name)
+            _pos = _path.find('/')
+            if _path[:_pos] in common.KNOWN_PRECISIONS:
+                _path = _precision_marker + _path[_pos:]
+            _paths.append(_path)
```

Comment:
This logic is redundant. For `dldt` models it's guaranteed that the files are `<precision>/<name>.xml` and `<precision>/<name>.bin`.

Reply:
It's used to replace it after to requested precision.

Reply:
replaced

Reply:
I'm not sure what you replaced, but it's still redundant. You don't need a loop or a `_paths` list. You know what the names of the files are.